Integrations/integration-C2sec.yml

```diff
@@ -0,0 +1,246 @@
+description: Understand Your Cyber Exposure as Easy as a Google Search
+configuration:
+- display: URL (e.g https://api.c2sec.com/api)
+  name: endpointURL
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: API Key
+  name: apikey
+  defaultvalue: ""
+  type: 4
+  required: true
+- display: Default Domain Name
+  name: domainName
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Trust any certificate (unsecure)
+  name: unsecure
+  defaultvalue: ""
+  type: 8
+  required: false
+script:
+  script: |-
+    import requests
+    import json
+    import datetime
+
+    # disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+
+    API_KEY = demisto.params()['apikey']
+    HEADERS = {'Content-Type': 'application/json', 'Accept': 'application/json'}
+    if 'domain' in demisto.args():
+      DOMAIN = demisto.args()['domain']
+    else:
+        DOMAIN = demisto.params()['domainName']
+    if demisto.params()['endpointURL'].count("/") == 3:
+      ENDPOINTURL = demisto.params()['endpointURL']+ "/"
+    else:
+        ENDPOINTURL = demisto.params()['endpointURL']
+    USE_SSL = not demisto.params().get('unsecure', False)
+
+    def addcompany(domain, newscan):
+       if newscan == 'true':
+         newscan = 1
+       else:
+           newscan = 0
+       params = {'apikey': API_KEY, 'domain': domain, 'newscan': newscan}
+       call = requests.get(ENDPOINTURL + 'iriskaddcompany', headers=HEADERS, verify=USE_SSL, params=params)
+       if call.status_code == requests.codes.ok:
+          md = tableToMarkdown('Add Company Result', call.json())
+          return  {
+                    'Type': entryTypes['note'],
+                    'ContentsFormat': formats['json'],
+                    'Contents': call.json(),
+                    'HumanReadable': md,
+                    'EntryContext': {"C2sec.AddCompany": call.json() }
+                }
+       else:
+            return ('Error adding a new company - status code: [%d] - reason: %s' % (call.status_code, call.text))
+
+
+
+    def queryWorkitem(workitemid):
+       params = {'apikey': API_KEY, 'workitemid': workitemid}
+       headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
+       call = requests.get(ENDPOINTURL + '/iriskqueryapistatus', headers=HEADERS, verify=USE_SSL, params=params)
+       if call.status_code == requests.codes.ok:
+         resp = call.json()
+         resp['apistatus']['workitemid'] = workitemid
+         md = tableToMarkdown('Query WorkItem Result', resp['apistatus'])
+         return  {
+                    'Type': entryTypes['note'],
+                    'ContentsFormat': formats['json'],
+                    'Contents': resp,
+                    'HumanReadable': md,
+                    'EntryContext': {"C2sec.Scan(val.workitemid && val.workitemid == obj.workitemid)": resp['apistatus']}
```

Comment:
what do we have in apistatus that is relevant to the scan itself?

Reply:
@ronykoz - this parameter is not added by default by their api so I need to track the status of the scan somehow hence that's why it's added into the context.